### ISSUE TITLE
Transform does not work with nested paths

### ISSUE THREAD

**Issue by jonathan-wilkinson** [Automattic/mongoose#9543]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

When using a transform on a nested path, the property is not transformed. Instead, a new dot-path style property is added to the root of the object.

**If the current behavior is a bug, please provide the steps to reproduce.**
```
const mongoose = require('mongoose');

const MyModel = mongoose.model(
    'MyModel',
    new mongoose.Schema({
        a: {
            b: {
                type: String,
                transform: () => 'x', // Always transform 'a.b' to 'x'
            },
        },
    }),
);

// Create instance of MyModel where path 'a.b' === 'c'
const instance = new MyModel({ a: { b: 'c' } });

/**
 * Expect
 * ```
 * { a: { b: 'x' } }
 * ```
 * See
 * ```
 * { a: { b: 'c' }, 'a.b': 'x' }
 * ```
 */
console.log(instance.toJSON());
```

**What is the expected behavior?**
Expect that
- the property associated with the transform is transformed
- no new properties are added to the object

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

node: v12.18.3
mongoose: 5.10.9
mongodb: 3.6.2

** Notes **

This behaviour is coming from this line: https://github.com/Automattic/mongoose/blob/c83999b3b792d540763fb0b2dd61173730fbac3d/lib/document.js#L3539 .

`path` can be a dot-path to the relevant property.

**Comment by jonathan-wilkinson** [comment]
Created a PR: https://github.com/Automattic/mongoose/pull/9544 . Not contributed here before so let me know if there's anything else I need to do.

**Comment by vkarpov15** [comment]
@jonathan-wilkinson thanks for reporting and for your PR, this fix will be in v5.10.15 :+1:
